lib/internal/errors.js

```diff
@@ -40,9 +41,45 @@ const prepareStackTrace = (globalThis, error, trace) => {
   if (trace.length === 0) {
     return errorString;
   }
-  return `${errorString}\n    at ${trace.join('\n    at ')}`;
+
+  if (experimentalSourceMaps === undefined) {
```

Comment:
Do we really need the `prepareStackTrace` polyfills to work with source maps? I think it would be fine if we choose a different implementation of `prepareStackTrace` during pre-execution and use that instead instead of doing a runtime check for every `error.stack` being accessed. At least we can probably leave those quirks out from the start and see if anyone actually wants them enough to send in a feature request.

Reply:
If I follow, you're suggesting simply swapping in a new `prepareStackTrace` here, if `--enable-source-maps` is set:

https://github.com/nodejs/node/blob/master/lib/internal/bootstrap/node.js#L316

And we'd skip adding:

```js
  // `globalThis` is the global that contains the constructor which
  // created `error`.
  if (typeof globalThis.Error.prepareStackTrace === 'function') {
    return globalThis.Error.prepareStackTrace(error, trace);
  }
```

etc., for the time being, let people ask for it?

Reply:
Yes, and we can add a note about this in the doc saying that `--enable-source-maps` does not work with the `Error.prepareStackTrace` hook (I think that would be reasonable)

Reply:
fwiw we still use prepareStackTrace in the repl and in assertions (and maybe when checking the buffer deprecation??)